microsoft/testsuites/ltp/ltp.py

```diff
@@ -124,37 +154,36 @@ def run_test(
             # write skip test to skipfile with newline separator
             skip_file_value = "\n".join(skip_tests)
             self.node.tools[Echo].write_to_file(
-                skip_file_value, PurePosixPath(self.LTP_SKIP_FILE), sudo=True
+                skip_file_value, PurePosixPath(self.skip_path), sudo=True
             )
-            parameters += f"-S {self.LTP_SKIP_FILE} "
+            parameters += f"-S {self.skip_path} "
 
         # Minimum 4M swap space is needed by some mmp test
         if self.node.tools[Free].get_swap_size() < 4:
             self.node.tools[Swap].create_swap()
 
         # run ltp tests
-        command = f"{self.command} {parameters}"
-        self.node.execute_async(
-            f"echo y | {command}",
+        self.run(
```

Comment:
Use the async process, so the timeout detection is used with pgrep. Because some SSH connection will be disconnected, if no output in 1 minute. If your scenario doesn't support pgrep, use an parameter to control it like set `ltp_use_pgrep_timeout` to false.

Reply:
From my debug, it seems there is an issue when self.node.execute_async() is called.  self.node.execute_async() will just start to run the command, but if username and password are needed before the command is actually running, the command will never run. However, if self.run() is called, it will not only call self.node.execute_async(), but also wait for the result. This wait helps deal with the username and password issue before the command is run. So if pgrep is needed, the username and password issue also needs to be solved. Please let me know your suggestion.

Reply:
Please check with @lubaihua33 to move the logic to shell level and call it in execute_async.

Reply:
Sure, I will do that, By the way, that change of execute_async() is to make sure the command will be exectuted anyway, however, the current implementation of pgrep.wait_process() just checks the command and exits when the command is not found, this might cause an issue too. For example, while execute_async() is dealing with username and password, pgrep.wait_process() might think the command is finished as it doesn't detect the process, thus the test is treated as done by mistake, and no test result is to be found, which will fail the test unfortunately. I think pgrep.wait_process() might need to check the start of the command first, and then checks when the command is finished. Please let me know your suggestion.

Reply:
The `execute_async` doesn't mean the method is fully async. The username/password checks are synchronous. Before the checks completed, the `execute_async` doesn't exit, so the pgrep won't start.

Reply:
that makes sense, thanks for clarification.

Reply:
You can make the change in next PR.